cmd/xl-storage.go

```diff
@@ -494,9 +497,15 @@ func (s *xlStorage) NSScanner(ctx context.Context, cache dataUsageCache, updates
 			}
 			return sizeSummary{}, errSkipFile
 		}
+
+		var versioned bool
+		if vcfg != nil && vcfg.PrefixEnabled(item.objectPath()) || vcfg.PrefixSuspended(item.objectPath()) {
```

Comment:
There is a missing parenthesis here for the OR condition

Reply:
Wait why would that be needed here ? @vadmeste since && will evaluate first.

Reply:
> Wait why would that be needed here ? @vadmeste since && will evaluate first.

if cfg is nil, then this will still be evaluated => vcfg.PrefixSuspended(item.objectPath()))

Reply:
@vadmeste that is not true - that's what I meant by `vcfg` being nil here https://go.dev/play/p/PZHbQSYrpze